onnx/defs/shape_inference.h

```diff
@@ -154,131 +141,141 @@ multiplyDims(const TensorShapeProto& shape, int from, int upto_exclusive) {
   return dim;
 }
 
-void propagateElemTypeWithValidation(
-    const TypeProto* input_type,
-    TypeProto* output_type);
+inline int32_t getTensorElementType(const TypeProto& type) {
```

Comment:
(General comment, not specific to this PR). I wonder if this include file is worth splitting into an include-file and implementation-file (in a separate PR, doesn't have to be in this). It is growing big, and it is better to reduce unnecessary implementation-dependence some users might acquire.

Reply:
I was thinking along the same lines.

Reply:
I also agree

Reply:
I think this would fall under refactoring category and we better manage it under a separate PR.